sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/SubscriptionPartitionProcessorImplTest.java

```diff
@@ -63,24 +60,23 @@
 import org.mockito.InOrder;
 import org.mockito.Mock;
 import org.mockito.Spy;
-import org.mockito.stubbing.Answer;
 
 @RunWith(JUnit4.class)
 @SuppressWarnings("initialization.fields.uninitialized")
 public class SubscriptionPartitionProcessorImplTest {
+  private static final SubscriptionPartition PARTITION =
+      SubscriptionPartition.of(example(SubscriptionPath.class), example(Partition.class));
+
   @Spy RestrictionTracker<OffsetByteRange, OffsetByteProgress> tracker;
   @Mock OutputReceiver<SequencedMessage> receiver;
-  @Mock Function<Consumer<List<SequencedMessage>>, Subscriber> subscriberFactory;
+  @Mock Supplier<MemoryBufferedSubscriber> subscriberFactory;
 
   @Rule public Timeout globalTimeout = Timeout.seconds(30);
 
-  abstract static class FakeSubscriber extends FakeApiService implements Subscriber {}
```

Comment:
I guess maybe the interface/implementation is following an existing pattern used with `Subscriber` here?

I still think it's worth considering if we actually need the new interfaces though

Reply:
Nack, see above.